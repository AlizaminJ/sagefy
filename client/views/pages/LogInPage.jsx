import React from 'react'
import { Link } from 'react-router-dom'
import { shape, string } from 'prop-types'
import Layout from '../components/Layout'
import Icon from '../components/Icon'
import FormErrorsTop from '../components/FormErrorsTop'
import FormErrorsField from '../components/FormErrorsField'

export default function LogInPage({
  hash,
  gqlErrors,
  body: { name },
  query: { redirect },
}) {
  return (
    <Layout
      hash={hash}
      page="LogInPage"
      title="Log In"
      description="Log in to Sagefy. Continue learning your subjects. And help us make new content to grow Sagefy."
    >
      <FormErrorsTop formErrors={gqlErrors} />
      <FormErrorsField formErrors={gqlErrors} field="all" />

      <section>
        <h1>
          Log in to Sagefy <Icon i="logIn" s="xxl" />
        </h1>

        <p>
          Don&apos;t have an account?{' '}
          <Link to={redirect ? `/sign-up?redirect=${redirect}` : '/sign-up'}>
            <Icon i="signUp" /> Sign Up
          </Link>
          .<br />
          Forgot your password?{' '}
          <Link to="/password">
            <Icon i="password" /> Reset
          </Link>
          .
        </p>

        <form action="" method="POST">
          <p>
            <label htmlFor="name">Name or Email</label>
            <input
              id="name"
              name="name"
              placeholder="example: Unicorn"
              type="text"
              size="40"
              autoFocus
              value={name}
            />
          </p>
          <FormErrorsField formErrors={gqlErrors} field="name" />
          <p>
            <label htmlFor="password">Password</label>
            <input
              id="password"
              name="password"
              type="password"
              size="40"
              required
              pattern=".{8,}"
            />
          </p>
          <FormErrorsField formErrors={gqlErrors} field="password" />
          <p>
            <button type="submit">
              <Icon i="logIn" /> Log In
            </button>
          </p>
        </form>
      </section>
    </Layout>
  )
}

LogInPage.propTypes = {
  hash: string.isRequired,
  gqlErrors: shape({}),
  body: shape({}),
  query: shape({}).isRequired,
}

LogInPage.defaultProps = {
  gqlErrors: {},
  body: {},
}
