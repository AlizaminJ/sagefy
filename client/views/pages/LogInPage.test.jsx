import LogInPage from './LogInPage'

describe('LogInPage', () => {
  it('should render the LogInPage', () => {
    expect(LogInPage({ gqlErrors: {}, body: {}, query: {} })).toMatchSnapshot()
  })
})
