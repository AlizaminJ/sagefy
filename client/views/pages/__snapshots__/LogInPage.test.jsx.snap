// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LogInPage should render the LogInPage 1`] = `
<Layout hash={hash}
  description="Log in to Sagefy. Continue learning your subjects. And help us make new content to grow Sagefy."
  page="LogInPage"
  title="Log In"
>
  <FormErrorsTop
    formErrors={Object {}}
  />
  <FormErrorsField
    field="all"
    formErrors={Object {}}
  />
  <section>
    <h1>
      Log in to Sagefy
      <Icon
        i="logIn"
        s="xxl"
      />
    </h1>
    <p>
      Don't have an account?

      <Link
        replace={false}
        to="/sign-up"
      >
        <Icon
          i="signUp"
          s="m"
        />
         Sign Up
      </Link>
      .
      <br />
      Forgot your password?

      <Link
        replace={false}
        to="/password"
      >
        <Icon
          i="password"
          s="m"
        />
         Reset
      </Link>
      .
    </p>
    <form
      action=""
      method="POST"
    >
      <p>
        <label
          htmlFor="name"
        >
          Name or Email
        </label>
        <input
          autoFocus={true}
          id="name"
          name="name"
          placeholder="example: Unicorn"
          size="40"
          type="text"
        />
      </p>
      <FormErrorsField
        field="name"
        formErrors={Object {}}
      />
      <p>
        <label
          htmlFor="password"
        >
          Password
        </label>
        <input
          id="password"
          name="password"
          pattern=".{8,}"
          required={true}
          size="40"
          type="password"
        />
      </p>
      <FormErrorsField
        field="password"
        formErrors={Object {}}
      />
      <p>
        <button
          type="submit"
        >
          <Icon
            i="logIn"
            s="m"
          />
           Log In
        </button>
      </p>
    </form>
  </section>
</Layout>
`;
